avocado/core/job.py

```diff
@@ -68,21 +72,48 @@ def register_job_options():
 
 
 class Job:
-
-    """
-    A Job is a set of operations performed on a test machine.
+    """A Job is a set of operations performed on a test machine.
 
     Most of the time, we are interested in simply running tests,
     along with setup operations and event recording.
+
+    A job has multiple test suites attached to it. Please keep in mind that
+    when creating jobs from the constructor (`Job()`), we are assuming that you
+    would like to have control of the test suites and you are going to build
+    your own TestSuites.
+
+    If you would like any help to create the job's test_suites from the config
+    provided, please use `Job.from_config()` method and we are going to do our
+    best to create the test suites.
+
+    So, basically, as described we have two "main ways" to create a job:a
+
+    1. Automatic discovery, using `from_config()` method::
+
+        job = Job.from_config(job_config=job_config,
+                              suites_configs=[suite_cfg1, suite_cfg2])
+
+    2. Manual or Custom discovery, using the constructor::
+
+        job = Job(config=config,
+                  test_suites=[suite1, suite2, suite3])
     """
 
-    def __init__(self, config=None):
-        """
-        Creates an instance of Job class.
+    def __init__(self, config=None, test_suites=None):
+        """Creates an instance of Job class.
+
+        Note that `config` and `test_suites` are optional, if not passed you
+        need to change this before running your tests. Otherwise nothing is
+        will run. If you need any help to create the test_suites from the
+        config, then use the `Job.from_config()` method.
 
         :param config: the job configuration, usually set by command
                        line options and argument parsing
         :type config: dict
+        :param test_suites: A list with TestSuite objects. If is None the job
+                            will have an empty list and you can add suites
+                            after init accessing job.test_suites.
```

Comment:
It is optional since the default is None. If you would like that you need to run `Job.from_config()`.

I tried to document the outcome of this discussion at the docstrings:

https://github.com/avocado-framework/avocado/pull/4041#discussion_r464644172

Reply:
What I was trying to mean is that the user will always need to do: 

`Job(config, test_suites)` 

or 

```
j = Job(config)
j.from_config(suites)
```

But won't be able to do:

`J(config)`

With the test suite into the config anymore.

That is the impression I have when I read this docstring.

Reply:
@willianrampazzo what about after reading the class Job() docstring?

Reply:
Also the __init__ says that... "...config and test_suties are optional, if not passed you need to change this before running your tests..."

Reply:
Okay, from previous discussion I thought we were going to support `Job(config)` with the test case into the `config`.

Reply:
@willianrampazzo yes, sorry about the confusion.

During the implementation, I was uncomfortable with some aspects that I tried to explain here https://github.com/avocado-framework/avocado/pull/4041#discussion_r464644172

Hopefully will be clear with the docstrings and documentation.

Reply:
Yah, I missed that comment in the past, sorry for the noise.